operator/bundle_helpers/patch-csv.py

```diff
@@ -163,15 +164,23 @@ def get_previous_y_stream(version):
 
 
 def parse_args():
-    parser = argparse.ArgumentParser(description='Patch StackRox Operator ClusterServiceVersion file')
+    partial_parser = argparse.ArgumentParser(add_help=False)
+    partial_parser.add_argument('--related-images-mode', nargs='?')
+    args, _ = partial_parser.parse_known_args()
+    if args.related_images_mode == 'help':
+        print_related_images_mode_help()
+        sys.exit(0)
```

Comment:
The approach looks overkill to me. It would be sufficient to just extend `help` directly in the `--related-images-mode`. I would not insist changing that if you as the author feel happy about this nested-help approach.

What concerns me though is the code duplication. Instantiating ArgumentParser and creating `--related-images-mode` parameter twice definitely smells like it will come with pains at later maintenance.
It looks possible to me to handle the nested help after the original `parser.parse_args()` call. Wouldn't that work?

Reply:
Regarding your suggestion at the end, no that wouldn't work (conveniently), because the user would have to pass all mandatory arguments (there are some) on the command line to get to the point where the help be displayed.

tbh, I wasn't able to come up with a *conscise*, yet readable, description of *three* different modes of operation for an in-line help text. Especially given that we are not able to control formatting (newlines, etc.) unless we would switch to some "raw" formatter for the argument parsing help rendering.

Reply:
If you have a concrete suggestion for a such a concise and understandable explanation to be embedded into the in-line help, I am happy to use that.

Otherwise I would be quite happy if we get this merged.

Reply:
Here's one example that does not resort to the second ArgumentParser. https://github.com/stackrox/stackrox/pull/12974

Reply:
Thank you, please merge.

Reply:
Here's a suggestion that stays within the current approach but reduces duplication around `add_argument('--related-images-mode', ...)` which concerned me. https://github.com/stackrox/stackrox/pull/12975

Reply:
Here's a third suggestion where help message is shortened https://github.com/stackrox/stackrox/pull/12976

Reply:
merged first suggestion, hence resolving.